DOCSUP-7099: edit and translate (runningConcurrency, max_parallel_replicas)

lehasm opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Documentation for #18508 and #21113

lehasm committed:
Перевод описания функции (WIP)

lehasm committed:
runningConcurrency() english description changed.

lehasm committed:
max_parallel_replicas description edited

lehasm committed:
en text improved.

olgarev commented on docs/en/operations/settings/settings.md:
```diff
@@ -1097,14 +1097,25 @@ See the section “WITH TOTALS modifier”.
-In addition, this setting will produce incorrect results when joins or subqueries are involved, and all tables don't meet certain conditions. See [Distributed Subqueries and max_parallel_replicas](../../sql-reference/operators/in.md#max_parallel_replica-subqueries) for more details.
+**Additional Info** 
+
+This setting is only useful for replicated tables with a sampling key. A query may be executed faster by executing on several servers in parallel. But query performance may degrade in some cases:
```
```suggestion
This setting is only useful for replicated tables with a sampling key. A query may be processed faster if it is executed on several servers in parallel. But query performance may degrade in some cases:
```

olgarev commented on docs/en/sql-reference/functions/other-functions.md:
````diff
@@ -911,29 +911,30 @@ Same as for [runningDifference](../../sql-reference/functions/other-functions.md
 ```
 
-**Arguments**
+An event has start time and end time. Columns with these times must have the same data type. The start time is included in an event, while the stop time is excluded. The function analyses each event start time and calculates the total number of events that are active including the starting event.
````
```suggestion
Each event has a start time and an end time. Columns with these two values must be of the same data type. The start time is included in the event, while the end time is excluded. The function calculates the total number of events that are active at the specified start time.
```

olgarev commented on docs/en/sql-reference/functions/other-functions.md:
```diff
@@ -911,29 +911,30 @@ Same as for [runningDifference](../../sql-reference/functions/other-functions.md
-Note that two columns `begin` and `end` must have the same type.
+**Arguments**
+
+-   `start` — A column with start time of events. [Date](../../sql-reference/data-types/date.md), [DateTime](../../sql-reference/data-types/datetime.md), or [DateTime64](../../sql-reference/data-types/datetime64.md).
+-   `end` — A column with end time of events.  [Date](../../sql-reference/data-types/date.md), [DateTime](../../sql-reference/data-types/datetime.md), or [DateTime64](../../sql-reference/data-types/datetime64.md).
```
```suggestion
-   `start` — A column with the start time of events. [Date](../../sql-reference/data-types/date.md), [DateTime](../../sql-reference/data-types/datetime.md), or [DateTime64](../../sql-reference/data-types/datetime64.md).
-   `end` — A column with the end time of events.  [Date](../../sql-reference/data-types/date.md), [DateTime](../../sql-reference/data-types/datetime.md), or [DateTime64](../../sql-reference/data-types/datetime64.md).
```

olgarev commented on docs/en/sql-reference/functions/other-functions.md:
```diff
@@ -942,31 +943,29 @@ Type: [UInt32](../../sql-reference/data-types/int-uint.md)
 Input table:
```
```suggestion
Consider the table:
```

lehasm committed:
Fixes after review applied

lehasm committed:
ru translation

olgarev commented on docs/ru/operations/settings/settings.md:
```diff
@@ -1087,8 +1087,22 @@ load_balancing = round_robin
+
+**Дополнительная информация**
+
+Эта настройка полезна для реплицируемых таблиц с ключом семплирования. Запрос может обрабатываться быстрее, если он выполняется на нескольких серверах параллельно. Однако производительность обработки запроса может наоборот упасть в некоторых ситуациях:
```
```suggestion
Эта настройка полезна для реплицируемых таблиц с ключом сэмплирования. Запрос может обрабатываться быстрее, если он выполняется на нескольких серверах параллельно. Однако производительность обработки запроса может наоборот упасть в некоторых ситуациях:
```

olgarev commented on docs/ru/operations/settings/settings.md:
```diff
@@ -1087,8 +1087,22 @@ load_balancing = round_robin
+
+Эта настройка полезна для реплицируемых таблиц с ключом семплирования. Запрос может обрабатываться быстрее, если он выполняется на нескольких серверах параллельно. Однако производительность обработки запроса может наоборот упасть в некоторых ситуациях:
+
+- Позиция ключа семплирования в ключе разбиения не позволяет выполнять эффективное сканирование.
+- Добавление ключа семплирования в таблицу делает фильтрацию по другим столбцам менее эффективной.
+- Ключ семплирования является выражением, которое сложно вычисляется.
+- У распределения задержек в кластере длинный хвост, из-за чего запросы к нескольким серверам увеличивают общее время задержки.
```
```suggestion
- Позиция ключа сэмплирования в ключе разбиения не позволяет выполнять эффективное сканирование.
- Добавление ключа сэмплирования в таблицу делает фильтрацию по другим столбцам менее эффективной.
- Ключ сэмплирования является выражением, которое сложно вычисляется.
- У распределения задержек в кластере длинный хвост, из-за чего запросы к нескольким серверам увеличивают общее время задержки.
```

olgarev commented on docs/ru/operations/settings/settings.md:
```diff
@@ -1087,8 +1087,22 @@ load_balancing = round_robin
+- У распределения задержек в кластере длинный хвост, из-за чего запросы к нескольким серверам увеличивают общее время задержки.
+
+!!! warning "Предупреждение"
+    Параллельное выполнение запроса приведет к неверному результату, когда в запросе есть объединение или подзапросы и при этом таблицы не удовлетворяют некоторым требованиям. Подробности смотрите в [Распределенные подзапросы и max_parallel_replicas](../../sql-reference/operators/in.md#max_parallel_replica-subqueries).
```
```suggestion
    Параллельное выполнение запроса может привести к неверному результату, если в запросе есть объединение или подзапросы, и при этом таблицы не удовлетворяют некоторым требованиям. Подробности смотрите в разделе [Распределенные подзапросы и max_parallel_replicas](../../sql-reference/operators/in.md#max_parallel_replica-subqueries).
```

olgarev commented on docs/ru/sql-reference/functions/other-functions.md:
```diff
@@ -852,6 +852,64 @@ WHERE diff != 1
+
+## runningConcurrency {#runningconcurrency}
+
+Определяет количество одновременно идущих событий в моменты начала событий. 
```
```suggestion
Определяет все моменты начала событий в указанном интервале и подсчитывает количество событий, происходящих одновременно в каждый из этих моментов. 
```

olgarev commented on docs/ru/sql-reference/functions/other-functions.md:
````diff
@@ -852,6 +852,64 @@ WHERE diff != 1
+runningConcurrency(start, end)
+```
+
+У каждого события есть время начала и время завершения. Столбцы с этими значениями должны содержать данные одинакового типа. Время начала включается в событие, а время завершения исключается из события. Для момента начала каждого события функция вычисляет количество идущих событий.
````
```suggestion

```

olgarev commented on docs/en/sql-reference/functions/other-functions.md:
````diff
@@ -911,62 +911,59 @@ Same as for [runningDifference](../../sql-reference/functions/other-functions.md
 ```
 
-**Arguments**
+Each event has a start time and an end time. Columns with these two values must be of the same data type. The start time is included in the event, while the end time is excluded. The function calculates the total number of active events for each start time in the specified period.
````
```suggestion

```

olgarev commented on docs/en/sql-reference/functions/other-functions.md:
```diff
@@ -911,62 +911,59 @@ Same as for [runningDifference](../../sql-reference/functions/other-functions.md
-    Events spanning multiple data blocks will not be processed correctly. The function resets its state for each new data block.
-
-The result of the function depends on the order of data in the block. It assumes the beginning time is sorted in ascending order.
+Calculates the number of events that are concurrent at event start time.
```
```suggestion
Each event has a start time and an end time. The start time is included in the event, while the end time is excluded. 

The function calculates the total number of active (concurrent) events for each start time in the specified period.

Columns with a start time and an end time must be of the same data type. 
```

olgarev commented on docs/ru/sql-reference/functions/other-functions.md:
```diff
@@ -852,6 +852,64 @@ WHERE diff != 1
+У каждого события есть время начала и время завершения. Столбцы с этими значениями должны содержать данные одинакового типа. Время начала включается в событие, а время завершения исключается из события. Для момента начала каждого события функция вычисляет количество идущих событий.
+
+!!! warning "Предупреждение"
+    Каждый блок данных обрабатывается независимо. Если события из разных блоков данных накладываются, то они не могут быть корректно обработаны.
```
```suggestion
    Каждый кусок данных обрабатывается независимо. Если события из разных кусков данных накладываются по времени, они не могут быть корректно обработаны.
```

olgarev commented on docs/ru/sql-reference/functions/other-functions.md:
```diff
@@ -852,6 +852,64 @@ WHERE diff != 1
+**Аргументы**
+
+-   `begin` — Столбец с временем начала событий. [Date](../../sql-reference/data-types/date.md), [DateTime](../../sql-reference/data-types/datetime.md), or [DateTime64](../../sql-reference/data-types/datetime64.md).
+-   `end` — Столбец с временем завершения событий.  [Date](../../sql-reference/data-types/date.md), [DateTime](../../sql-reference/data-types/datetime.md), or [DateTime64](../../sql-reference/data-types/datetime64.md).
```
```suggestion
-   `end` — Столбец с временем окончания событий.  [Date](../../sql-reference/data-types/date.md), [DateTime](../../sql-reference/data-types/datetime.md) или [DateTime64](../../sql-reference/data-types/datetime64.md).
```

olgarev commented on docs/ru/sql-reference/functions/other-functions.md:
```diff
@@ -852,6 +852,64 @@ WHERE diff != 1
+
+**Аргументы**
+
+-   `begin` — Столбец с временем начала событий. [Date](../../sql-reference/data-types/date.md), [DateTime](../../sql-reference/data-types/datetime.md), or [DateTime64](../../sql-reference/data-types/datetime64.md).
```
```suggestion
-   `begin` — Столбец с временем начала событий. [Date](../../sql-reference/data-types/date.md), [DateTime](../../sql-reference/data-types/datetime.md) или [DateTime64](../../sql-reference/data-types/datetime64.md).
```

olgarev commented on docs/ru/sql-reference/functions/other-functions.md:
```diff
@@ -852,6 +852,64 @@ WHERE diff != 1
+
+**Возвращаемое значение**
+
+-   Количество одновременно идущих событий в момента начала каждого события.
```
```suggestion
-   Количество одновременно идущих событий на каждый момент начала события в указанном интервале.
```

olgarev commented on docs/ru/sql-reference/functions/other-functions.md:
```diff
@@ -852,6 +852,64 @@ WHERE diff != 1
+
+**Пример**
+
+Для таблицы:
```
```suggestion
Рассмотрим таблицу:
```

lehasm committed:
Function description placed before syntax.
In in.md created header max_parallel_replica-subqueries with empty text. It is link target from runningConcurrency

lehasm committed:
Setting description minor changes

lehasm committed:
Warning moved close to function description
Order changed inside warning block

lehasm committed:
Fixed link in runningDifferenceStartingWithFirstValue

lehasm committed:
Fixed link in runningDifferenceStartingWithFirstValue

lehasm committed:
Fixed link in runningDifferenceStartingWithFirstValue

lehasm committed:
fix file name in link

lehasm committed:
more fix link

lehasm commented:
There are some conflicts. I will try to resolve them a bit later.

den-crane commented:
I see it removes these sentences

**In limited circumstances, this can make a query faster by executing it on more servers. ... There are cases where performance will not improve or even worsen:**

It's actually quite important. max_parallel_replicas is mostly unusable and in 99% cases makes performance worse.

lehasm committed:
Merge with master

lehasm committed:
Merge branch 'master' into alexey-sm-DOCSUP-7099-translate-runningConcurrency

lehasm commented:
> In limited circumstances, this can make a query faster by executing it on more servers. ... There are cases where performance will not improve or even worsen:
It changed to _A query may be processed faster if it is executed on several servers in parallel. But the query performance may degrade in some cases:_ in _Additional Info_ section.
@den-crane , is that ok?

den-crane commented:
@lehasm I have a concern about 'some cases', because it's like 99.999% cases.

lehasm commented:
We can change it to^
_A query may be processed faster if it is executed on several servers in parallel. But there are *a lot of cases* when query performance may degrade:_
@den-crane, do you agree?

lehasm committed:
some cases changed to following cases

lehasm committed:
Merge branch 'alexey-sm-DOCSUP-7099-translate-runningConcurrency' of https://github.com/lehasm/ClickHouse into alexey-sm-DOCSUP-7099-translate-runningConcurrency

lehasm commented:
Changed _some cases_ to _following cases_ (neutral variant in my opinion).

den-crane commented:
ok
